### PR TITLE
Adding a static repository layer to the rest-java-module.

### LINKED ISSUE #7499: HIP-857: Add repository layer in the rest-java module
Add a basic repository layer along with tests to the rest-java module for the `allowances/nfts `query.

### PR DESCRIPTION
**Pull request by mgoelswirlds**
**Description**:
These are static queries for nft allowance rest API.

This PR modifies 
-  Adds static queries for the rest-java repo.
-  Edits the openapi yaml to fix formatting for the NFT allowances API.

**Related issue(s)**:
Fixes #7499 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/repository/NftAllowanceRepository.java [hunk 1]
```diff
@@ -18,6 +18,19 @@
...
 import com.hedera.mirror.common.domain.entity.AbstractNftAllowance.Id;
 import com.hedera.mirror.common.domain.entity.NftAllowance;
+import java.util.List;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 
-public interface NftAllowanceRepository extends CrudRepository<NftAllowance, Id> {}
+public interface NftAllowanceRepository extends CrudRepository<NftAllowance, Id> {
+
+    @Query(
```
Comment (mgoelswirlds) [inline hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/repository/NftAllowanceRepository.java:27]: These static queries do not cover all scenarios. Once we have the end to end flow for the API working. We will work on creating dynamic queries.

**Review by jascks — APPROVED** [review]
Seems to work as a stop gap measure to proceed with #7500 and later replace or supplement with #7690. LGTM.

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/repository/NftAllowanceRepository.java [hunk 2]
```diff
@@ -18,6 +18,29 @@
...
-public interface NftAllowanceRepository extends CrudRepository<NftAllowance, Id> {}
+public interface NftAllowanceRepository extends CrudRepository<NftAllowance, Id> {
+
+    /**
+     * This repository method will query based on the spender and further filter by the owner and token_id, order by owner first and then the token_id
+     * 1. It only covers owner eq and token_id gte scenario.
+     */
+    @Query(
+            value =
+                    "select * from nft_allowance where spender = ?  and  owner = ? and token_id >= ? order by owner,token_id limit ?",
```
Comment (steven-sheehy) [inline hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/repository/NftAllowanceRepository.java:33]: Probably this is the most common static query to have. If they omit owner or token_id you can default them to 0. If they omit direction you can default to ASC. If they omit limit you can default to 25. If they supply an `account.id=gte:X` you can subtract 1 from X. Similar for `token.id=gte:Y`.

```suggestion
                    "select * from nft_allowance where spender = ? and owner > ? and token_id > ? order by owner ?, token_id ? limit ?",
```

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/repository/NftAllowanceRepository.java [hunk 3]
```diff
@@ -18,6 +18,29 @@
...
+            nativeQuery = true)
+    List<NftAllowance> findByOwnerAndTokenEq(long accountId, long owner, long tokenId, int limit);
+
+    /**
+     * This repository method will query based on the owner and further filter by the spender and token_id, order by spender first and then the token_id
+     * 1. It only covers owner eq and token_id eq scenario.
+     */
+    @Query(
+            value =
+                    "select * from nft_allowance where owner = ? and spender = ? and token_id = ? order by spender,token_id limit ?",
```
Comment (steven-sheehy) [inline hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/repository/NftAllowanceRepository.java:43]: Probably query by owner=true would be more common than this.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 4]
```diff
@@ -239,7 +239,8 @@ paths:
     get:
       summary: This API is currently not supported.Get non fungible token allowances for an account
       description: |
-        ** This API is currently under development and is not supported yet **
+        **This API is currently under development and is not supported yet**
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:242]: Are you sure this fixes the problem?

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java [hunk 5]
```diff
@@ -28,9 +28,42 @@ class NftAllowanceRepositoryTest extends RestJavaIntegrationTest {
...
-    void save() {
+    void findBySpenderTokenGte() {
         var nftAllowance = domainBuilder.nftAllowance().get();
         nftAllowanceRepository.save(nftAllowance);
-        assertThat(nftAllowanceRepository.findById(nftAllowance.getId())).get().isEqualTo(nftAllowance);
+        assertThat(nftAllowanceRepository
+                        .findByOwnerAndTokenEq(
+                                nftAllowance.getSpender(), nftAllowance.getOwner(), nftAllowance.getTokenId(), 1)
+                        .get(0))
+                .isEqualTo(nftAllowance);
```
Comment (steven-sheehy) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java:38]: Same for others. 
```suggestion
        assertThat(nftAllowanceRepository
                        .findByOwnerAndTokenEq(
                                nftAllowance.getSpender(), nftAllowance.getOwner(), nftAllowance.getTokenId(), 1))
                .containsExactly(nftAllowance);
```

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java [hunk 6]
```diff
@@ -28,9 +28,42 @@ class NftAllowanceRepositoryTest extends RestJavaIntegrationTest {
     private final NftAllowanceRepository nftAllowanceRepository;
 
     @Test
-    void save() {
+    void findBySpenderTokenGte() {
         var nftAllowance = domainBuilder.nftAllowance().get();
         nftAllowanceRepository.save(nftAllowance);
```
Comment (steven-sheehy) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java:33]: Should save multiple to test limit

File: hedera-mirror-rest/api/v1/openapi.yml (hunk 4 above)
Comment (mgoelswirlds) [reply hedera-mirror-rest/api/v1/openapi.yml:242]: Yes. In intellij , I viewed it to verify.

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java [hunk 7]
```diff
@@ -21,16 +21,77 @@
...
     private final NftAllowanceRepository nftAllowanceRepository;
 
     @Test
-    void save() {
+    void findBySpenderAndFilterByOwnerGtAndTokenGt() {
         var nftAllowance = domainBuilder.nftAllowance().get();
         nftAllowanceRepository.save(nftAllowance);
-        assertThat(nftAllowanceRepository.findById(nftAllowance.getId())).get().isEqualTo(nftAllowance);
+        nftAllowanceRepository.save(domainBuilder.nftAllowance().get());
+        nftAllowanceRepository.save(domainBuilder.nftAllowance().get());
```
Comment (steven-sheehy) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/repository/NftAllowanceRepositoryTest.java:39]: Same for all other calls to save.
```suggestion
        domainBuilder.nftAllowance().persist();
        domainBuilder.nftAllowance().persist();
```

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by jascks — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by jascks — APPROVED** [review]
LGTM

**Pull request merged**